rts/CMakeLists.txt

```diff
@@ -85,9 +84,13 @@ if    (UNIX AND NOT MINGW)
 	endif(CMAKE_SYSTEM_NAME MATCHES "FreeBSD" OR CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
 endif (UNIX AND NOT MINGW)
 
-find_package_static(ZLIB REQUIRED)
-list(APPEND engineCommonLibraries ${IL_LIBRARIES} ${JPEG_LIBRARY} ${PNG_LIBRARY} ${TIFF_LIBRARY} ${GIF_LIBRARY})
-list(APPEND engineCommonLibraries 7zip prd::jsoncpp ${SPRING_MINIZIP_LIBRARY} ${ZLIB_LIBRARY} Tracy::TracyClient)
+find_package_static(ZLIB 1.2.7 REQUIRED)
+list(APPEND engineCommonLibraries DevIL::IL)
+
+if (NOT WIN32)
+	list(APPEND engineCommonLibraries PNG::PNG TIFF::TIFF JPEG::JPEG GIF::GIF)
```

Comment:
I think this will break in case we try to do shared linking on linux, e.g. when doing local build linking to local shared libs, won't it? Wouldn't it be best to have `DevIL::IL` always pull in correct dependencies? e.g. create another "virtual" target?

I think that having all the mess related to dependency distribution in single place, where dependencies are resolved, and not spread around code base as it's at the moment would make things cleaner.

Reply:
Woops, you're right, I fixed it.

> Wouldn't it be best to have DevIL::IL always pull in correct dependencies? e.g. create another "virtual" target?

We could hack it now (write a custom FIndDevIL.cmake or add the logic in ourselves). A package manager like conan would handle all of these problems. I guess the question is whether we want to wait for a conan solution or not -- what are your thoughts? I can adjust this PR either way.

Reply:
In my opinion, regardless if package manager is used or not, we can make CMake configs make sense and not mix many layers.

Modern build systems are separating external dependency resolution from build target configuration.
- bazel: external dependencies are in special `WORKSPACE` file, (in new versions bzlmod) and main build files just operate on targets
- a bunch of languages only specifies the dependencies separately and build files are "automatic" like go
- When you use package managers with cmake you are supposed to do the same

And in my opinion we should strive to do the same here: Dependency resolution be centralized in single place somewhere at the top, inclusion of the vendored project following, the rest of cmake files in engine should not care as much as possible about dependency resolution, whatever something is static or not, just reference correct target as dependency based on which header files the source includes.

If we do it this way, if we want to switch to package manager, or a different one in the future, one has to change *only* the cmake part responsible for dependency resolution, and not have to go through the whole project looking for where things are resolved.

So for example here, I believe that having the `if (PREFER_STATIC_LIBS)` is breaking this rule because source is not linking those libraries directly, it should only care about linking `DevIL::IL`. Yes, preferably the find script for DevIL would handle that (**Important**: I'm not experienced with this part of cmake myself yet that much, so take below workaround with grain of salt, maybe fixin find is *much* easier and proper), but it can be worked around e.g. by doing something like:

```cmake
add_library(recoil::DevIL INTERFACE)
find_package_static(DevIL)
target_link_libraries(recoil::DevIL DevIL::IL)
if    (PREFER_STATIC_LIBS)
	# dependencies of DevIL
	find_package_static(JPEG REQUIRED)
	find_package_static(PNG REQUIRED)
	find_package_static(TIFF REQUIRED)
	find_package_static(GIF REQUIRED)
	target_link_libraries(recoil::DevIL PNG::PNG TIFF::TIFF JPEG::JPEG GIF::GIF)
endif (PREFER_STATIC_LIBS)
```

and then just link `recoil::DevIL` where it's needed.

Reply:
I agree -- the behavior here (was and) is a hack.

Here is my plan:

Write a `FindDevIL.cmake` as the one included in cmake is not complex and I could easily modify to find the required dependency libraries. In that script I can carry the dependency imported targets as transitive targets (we won't have to target_link_libraries). This will work as is without a package manager.

A package manager like conan will generate CMake configs for targets, and default CMake to preferring configs over find modules. It's a bit subtle but what that means is that in the end, if something like conan is enabled, CMake will ignore the `FindDevIL.cmake` and use the conan-provided config. This is what we want since the conan-provided config will ensure that the dependencies are also linked in. Therefore I think this solution will work with and without conan.

Reply:
Sounds good, do you want to do it as part of this PR, or merge this first and do it in next ones? Up to you

Reply:
Let me fix it in this PR, I think it shouldn't take too long

Reply:
Ok check it out now, it should be more isolated